src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
I don't know how to search using only a part of a hash right now, but I don't see why it can not become possible. Thus, to be on the safe side hash it again, please.